File: test/Z.Core.Test/System.TimeSpan/TimeSpan.UtcFromNow.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Z.Core.Test
{
    [TestClass]
    public class System_TimeSpan_UtcFromNow
    {
        [TestMethod]
        public void UtcFromNow()
        {
            // Type
            var @this = new TimeSpan(1, 0, 0, 0);

            // Examples
            DateTime value = @this.UtcFromNow(); // return yesterday.

            // Unit Test
            Assert.IsTrue(DateTime.UtcNow.Subtract(value).Days <= -1);
        }
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.TimeSpan/TimeSpan.UtcFromNow.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library


Public Module Extensions_609

	''' <summary>
	'''     A TimeSpan extension method that add the specified TimeSpan to the current UTC (Coordinated Universal Time)
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>The current UTC (Coordinated Universal Time) with the specified TimeSpan added to it.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function UtcFromNow(this As TimeSpan) As DateTime
		Return DateTime.UtcNow.Add(this)
	End Function
End Module


